### ISSUE TITLE
508-defect-1 ❗️[COGNITION, SCREEN READER]: Screen reader users MUST know there is an alert after making a selection

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#19660]
# [508-defect-1](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a screen reader user, I need to know that there is an alert after specific radio button selections. Since I may be accustomed to navigating forms using `tab` , I may accidentally skip relevant information.

## Details

Upon selecting "no" on the supporting evidence page, an alert will appear notifying the user that they need to submit their service treatment records as soon as possible. This is not announced to screen reader users and may be skipped over should they use `tab`.

## Acceptance Criteria

- [ ] The alert is announced to the screen reader user

## Environment

* Operating System: any
* Browser: any
* Screenreading device: any
* Server destination: production

## Steps to Recreate

1. Reach step 3 of 5: Supporting evidence
2. Start your screen reader device of choice
2. Select "no"
3. Confirm no announcement is made

## Solution 

Two methods could be used to address this:
1. Add the alert into a live region that exists on the page at render
2. Use `aria-describedby` to link the alert to the "no" selection

### Live Region (Suggested)

An empty live-region could be added to exist on the page at render. Upon selecting "no" it can be populated with the alert, which would then be announced to the user. This may not be technically feasible with the form system, but would be a more appropriate use of aria in this scenario. It also better ensures that the user will hear the alert.

<img width="1397" alt="Screen Shot 2021-02-09 at 11 14 03 AM" src="https://user-images.githubusercontent.com/14154792/107393011-46dfaf00-6ac8-11eb-8550-12e36fa727d9.png">

### Aria-describedby (Band aid)

An easier approach may be to use `aria-describedby` to link the "no" selection with the alert. However, users may not hear the alert should they tab to the next selection quickly without listening to the full screen reader dialogue as the `aria-describedby` will only come _after_ the selection text and legend is read out. Since the legend is verbose on its own, there's a possibility users will tab to the next item before hearing the alert.

https://user-images.githubusercontent.com/14154792/107394140-88248e80-6ac9-11eb-8cc2-7c55712bf97c.mov
<img width="1532" alt="Screen Shot 2021-02-09 at 11 19 32 AM" src="https://user-images.githubusercontent.com/14154792/107394275-b1ddb580-6ac9-11eb-86f9-6a449ab99541.png">

## WCAG or Vendor Guidance (optional)

* [WCAG 2.1 Level A - 1.3.2 Meaningful Sequence](https://www.wuhcag.com/meaningful-sequence/) - Users who rely on assistive technology (such as a screen reader) to interpret content, require content to be presented in a meaningful order. If this is presented out of sequence, users may become disorientated and will not understand the content.
* [Gov.uk's check a service is eligible pattern](https://design-system.service.gov.uk/patterns/check-a-service-is-suitable/) would be a significantly more accessible and usable pattern to address this need. Until more testing is conducted with screen readers, the wizard pattern should be used cautiously.

## Screenshots or Trace Logs
Upon selecting "no", the screen reader should read out the information using `aria-describedby` 
<img width="762" alt="Screen Shot 2021-01-26 at 9 55 26 AM" src="https://user-images.githubusercontent.com/14154792/105863457-eaa46780-5fbe-11eb-96c1-b428bd3d9bf5.png">

**Comment by joshkimux** [comment]
[Bumping this up to a 508-defect-1 according to our documented guidelines](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Comment by joshkimux** [comment]
https://user-images.githubusercontent.com/14154792/113580740-20d01a00-95f4-11eb-85b5-6ec571d1ddca.mov

This has been fixed and will provide a band-aid solution as documented in the ticket! Since there's still the possibility screen reader users (particularly advanced ones) will not listen to the full announcement, I'll create a separate `defect-level-3` for converting this into an `aria-live` region as originally recommended.

Great work @Mottie 🥳 


https://user-images.githubusercontent.com/14154792/113580951-6db3f080-95f4-11eb-93ce-c791a3d9fcbd.mov
